rdflib/plugins/memory.py

```diff
@@ -38,15 +38,29 @@ def __init__(self, configuration=None, identifier=None):
         self.__namespace = {}
         self.__prefix = {}
 
+        # number of triples in the store
+        self.__size = 0
+
     def add(self, triple, context, quoted=False):
         """\
         Add a triple to the store of triples.
+
+        :Returns: 
+            1 in case of a new addition, 0 otherwise
         """
         # add dictionary entries for spo[s][p][p] = 1 and pos[p][o][s]
         # = 1, creating the nested dictionaries where they do not yet
         # exits.
         subject, predicate, object = triple
         spo = self.__spo
+        try:
+            # Triple already in store
+            _ = spo[subject][predicate][object]
```

Comment:
What is the performance impact here?

Reply:
As far as I [know](https://stackoverflow.com/questions/1963507/time-complexity-of-accessing-a-python-dict), on an average, it should be O(1).